src/extension.ts

```diff
@@ -207,17 +207,45 @@ async function autoUpdate() {
     }
 }
 
-function diagnosticCounter() {
+function progressCounter() {
+    const runningProgress: { [index: string]: boolean } = {};
+    const asPercent = (fraction: number): string => `${Math.round(fraction * 100)}%`;
     let runningDiagnostics = 0;
     lc.onReady().then(() => {
-        lc.onNotification(new NotificationType('rustDocument/beginBuild'), function(_f: any) {
+
+        stopSpinner('RLS');
+
+        lc.onNotification(new NotificationType('window/progress'), function (progress: any) {
+            if (progress.done) {
+                delete runningProgress[progress.id];
+            } else {
+                runningProgress[progress.id] = true;
+            }
+            if (Object.keys(runningProgress).length) {
+                let status = '';
+                if (typeof progress.percentage === 'number') {
+                    status = asPercent(progress.percentage);
+                } else if (progress.message) {
+                    status = progress.message;
+                } else if (progress.title) {
+                    status = `[${progress.title.toLowerCase()}]`;
```

Comment:
nit: I know it's up to clients to display this on the UI side, but shouldn't we just display it how servers send it?

Reply:
Ultimately I don't really want to end up in this part of the logic. This is a last resort fallback. We should try to send percentages or messages instead. It looked really bad to me with the caps, but this just taste and style, so I'm only going to argue my case once :)

If you think it's an important point, I'll change it.